modules/battery_manager/php/battery_manager.class.inc

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * This file contains the NDB_Menu_Filter_battery_manager class
+ *
+ * PHP Version 7
+ *
+ * @category Module
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @author   Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\battery_manager;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * Main class for battery manager module corresponding to /battery_manager/ URL
+ * Admin section of the LorisMenu.
+ *
+ * Displays a list of records in the test battery and control panel to search them
+ * Allows user to add, activate, and deactivate entries in the test battery
+ *
+ * PHP Version 7
+ *
+ * @category Module
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @author   Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Battery_Manager extends \NDB_Menu_Filter
+{
+    public $AjaxModule   = true;
+    public $skipTemplate = true;
+
+    /**
+     * Returns true if user has access to this page.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return $user->hasPermission('battery_manager_view');
+    }
+
+    /**
+     * This acts as an Ajax enpoint that handles all action requests from the 
+     * Battery Manager Module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db = \Database::singleton();
+        $user = $request->getAttribute('user');
+
+        $action = $request->getQueryParams()['action'];
```

Comment:
This seems to be a legacy from it being a script in the ajax directory. Now that it's implemented as a RequestHandler, it can include proper RESTful end points such as

`PUT battery_manager/batteries/$batteryname/$testname` in order to add or activate the test based on the status, or `DELETE battery_manager/batteries/$batteryname/$testname` to deactivate.

(I know batteries don't currently have name, but whatever name or id you're using to identify it this way..)

Reply:
I'm not sure if I completely understand. How would I send `PUT` and `DELETE` commands?